editor-extensions/vscode/reason-markdown-codeblock.json

```diff
@@ -1,15 +1,32 @@
 {
   "fileTypes": [],
-  "injectionSelector": "L:markup.fenced_code.block.markdown",
+  "injectionSelector": "L:text.html.markdown",
   "patterns": [
     {
       "include": "#reason-code-block"
     }
   ],
   "repository": {
     "reason-code-block": {
-      "begin": "(reason|reasonml)(\\s+[^`~]*)?$",
-      "end": "(^|\\G)(?=\\s*[`~]{3,}\\s*$)",
+      "name": "markup.fenced_code.block.markdown",
+      "begin": "(^|\\G)(\\s*)(\\`{3,}|~{3,})\\s*(?i:(re|reason|reasonml)(\\s+[^`~]*)?$)",
+      "end": "(^|\\G)(\\2|\\s{0,3})(\\3)\\s*$",
+      "beginCaptures": {
```

Comment:
Can you explain what all of these changes are doing? I would have expected that you'd only need to change line 11 to include `re|`

Reply:
@jaredly have you read my issues on vscode repo? https://github.com/microsoft/vscode/issues/75176
also this change: https://github.com/mjbvz/vscode-fenced-code-block-grammar-injection-example/commit/e333677e840fd28f382935650553736aaebc2207